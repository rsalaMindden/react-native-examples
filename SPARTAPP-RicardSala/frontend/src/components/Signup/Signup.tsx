import React from 'react';
import {SafeAreaView, Button, Image, Text, TextInput} from 'react-native';
import globalStyles from '../../theme/globalTheme';

const Signup = ({navigation}: any) => {
  return (
    <SafeAreaView style={globalStyles.backgound}>
      <Image
        style={globalStyles.logo__Standar}
        source={require('../../images/spartappYELLOW.png')}
      />
      <Text> Signup page</Text>
      <TextInput placeholder="User name" />
      <TextInput placeholder="password" />
      <Button title="Go to home" onPress={() => navigation.push('HomePage')} />
    </SafeAreaView>
  );
};

export default Signup;
