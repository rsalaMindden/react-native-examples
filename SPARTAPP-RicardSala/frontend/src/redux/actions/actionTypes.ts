export default {
  LOG_IN: 'LOG_IN',
  LOG_IN_ERROR: 'LOG_IN_ERROR',
  LOG_USER: 'LOG_USER',
  SERVER_ERROR: 'SERVER_ERROR',
  REGISTER_USER: 'REGISTER_USER',
  REGISTER_ERROR: 'REGISTER_ERROR',
};
