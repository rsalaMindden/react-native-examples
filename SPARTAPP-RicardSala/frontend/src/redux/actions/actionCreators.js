import actionTypes from './actionTypes';
import {env} from '../../../.env';
import axios from 'axios';
import {connect} from 'react-redux';

export function login(email, password) {
  return async dispatch => {
    try {
      const {data} = await axios.post(env.LOG_IN_URL, {email, password});
      dispatch({
        type: actionTypes.LOG_IN,
        user: data,
      });
    } catch (error) {
      dispatch({
        type: actionTypes.LOGIN_ERROR,
      });
    }
  };
}
