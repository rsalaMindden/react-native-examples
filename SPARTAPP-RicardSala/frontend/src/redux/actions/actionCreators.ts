import actionTypes from './actionTypes';
import {LOG_IN_URL} from '@env';
import {SING_UP_URL} from '@env';
import axios from 'axios';
export function loginUserAction(user: any) {
  return async (dispatch: any) => {
    user.email = user.email.toLowerCase();
    try {
      const {data} = await axios.post(LOG_IN_URL, user);
      console.log(data);
      dispatch({
        type: actionTypes.LOG_IN,
        user: data,
      });
    } catch (error: any) {
      dispatch({
        type: actionTypes.LOG_IN_ERROR,
      });
    }
  };
}
export function registerUser(user: any) {
  return async (dispatch: any) => {
    user.email = user.email.toLowerCase();
    console.log(user);
    try {
      const {data} = await axios.post(SING_UP_URL, user);
      console.log(data);
      dispatch({
        type: actionTypes.LOG_USER,
        data,
      });
    } catch (error: any) {
      if (error?.response?.status === 500) {
        dispatch({
          type: actionTypes.REGISTER_ERROR,
        });
      } else {
        dispatch({
          type: actionTypes.SERVER_ERROR,
        });
      }
    }
  };
}
