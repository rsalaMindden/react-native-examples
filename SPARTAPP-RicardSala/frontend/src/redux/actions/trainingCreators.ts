import actionTypes from './actionTypes';
import {LOAD_TRAININGS} from '@env';
import axios from 'axios';

export function loadTrainings() {
  return async (dispatch: any) => {
    try {
      const {data} = await axios.get(LOAD_TRAININGS);

      dispatch({
        type: actionTypes.LOAD_TRAININGS,
        data,
      });
    } catch (error: any) {
      dispatch({
        type: actionTypes.LOAD_TRAININGS_ERROR,
      });
    }
  };
}
