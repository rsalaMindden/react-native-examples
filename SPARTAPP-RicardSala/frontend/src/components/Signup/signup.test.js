import React from "react";
import Signup from './Signup'
import {render,fireEvent}from '../../utils/tetsUtils/test.utils'
import {registerUser} from '../../redux/actions/actionCreators';

const navigation={
    pop:jest.fn(),
    navigate:jest.fn()
}
jest.mock('../../redux/actions/actionCreators',()=>({
    registerUser: jest.fn()
  }));


  handleUsernameFocus=jest.fn()

describe('Given a Signup component',()=>{
    describe('When it is render',()=>{
        let screen
        beforeEach(()=>{
            screen=render(<Signup navigation={navigation}/>)
        })
        test('Then should match the snapshot',()=>{
            expect(screen).toMatchSnapshot();
          })

          describe('And you click on loginPageButton', () => {
            test('Then navigation.pop should have been called', () => {
              const loginPageButton = screen.getByTestId('loginPageButton');
              fireEvent.press(loginPageButton);
              expect(navigation.pop).toHaveBeenCalled();
            });
          })

          describe('And usernameInput is empty', () => {
            describe('And you click registerButton', () => {
              beforeEach(() => {
                const registerButton = screen.getByTestId('registerButton');
                fireEvent.press(registerButton);
              });
              test('Then should render a Text with testId invalidUsername', () => {
                expect(screen.queryByTestId('invalidUsername')).not.toBe(null);
              });
    })
})
describe('And you focus on emailInput', () => {
    test('Then should not render a Text with testId invalidUsername', () => {
      const usernameInput = screen.getByTestId('usernameInput');
      fireEvent(usernameInput, 'onChangeText');
      expect(screen.queryByTestId('invalidUsername')).toBe(null);
    });
  });
})
})

describe('And name is valid but email is not', () => {
  describe('And you click registerButton', () => {
    let screen;
    beforeEach(() => {
      screen=render(<Signup navigation={navigation}/>)
      const usernameInput = screen.getByTestId('usernameInput');
      fireEvent(usernameInput, 'focus');
      fireEvent.changeText(usernameInput, 'Riki');
      const registerButton = screen.getByTestId('registerButton');
      fireEvent.press(registerButton);
    });

    test('Then should render a Text with testId invalidEmail', () => {
      expect(screen.queryByTestId('invalidEmail')).not.toBe(null);
    });

    describe('And you focus on emailInput', () => {
      test('Then should not render a Text with testId invalidEmail', () => {
        const emailInput = screen.getByTestId('emailInput');
        fireEvent(emailInput, 'focus');
        expect(screen.queryByTestId('invalidEmail')).toBe(null);
      });
    });
  });
});
describe('And name, email are valid but email is not', () => {
  describe('And you click registerButton', () => {
    let screen;
    beforeEach(() => {
      screen=render(<Signup navigation={navigation}/>)
      const usernameInput = screen.getByTestId('usernameInput');
      const emailInput = screen.getByTestId('emailInput');
      fireEvent.changeText(usernameInput, 'Riki');
      fireEvent(emailInput, 'focus');
      fireEvent.changeText(emailInput, 'example@example.com');
      const registerButton = screen.getByTestId('registerButton');
      fireEvent.press(registerButton);
    });

    test('Then should render a Text with testId invalidPassword', () => {
      expect(screen.queryByTestId('invalidPassword')).not.toBe(null);
    });

    describe('And you focus on passwordInput', () => {
      test('Then should not render a Text with testId invalidPassword', () => {
        const passwordInput = screen.getByTestId('passwordInput');
        fireEvent(passwordInput, 'focus');
        expect(screen.queryByTestId('invalidPassword')).toBe(null);
      });
    });
  });
});

describe('And name, email, password, are valid', () => {
  describe('And you click registerButton', () => {
    const name = 'Riki';
    const email = 'juan@email.com';
    const password = 'Aa111111';
   
    let screen;
    beforeEach(() => {
      screen=render(<Signup navigation={navigation}/>)
       registerUser.mockReturnValue({ type: '' });
      const usernameInput = screen.getByTestId('usernameInput');
      const emailInput = screen.getByTestId('emailInput');
      const passwordInput = screen.getByTestId('passwordInput');
      fireEvent.changeText(usernameInput, name);
      fireEvent.changeText(emailInput, email);
      fireEvent.changeText(passwordInput, password);
      const registerButton = screen.getByTestId('registerButton');
      fireEvent.press(registerButton);
    });
    test('Then should call registerUser with the name, email and password typed', () => {
      expect(registerUser).toHaveBeenCalledWith({ name, email, password });
    });

    
    test('Then should call navigation.pop', () => {
      expect(navigation.pop).toHaveBeenCalled();
    });
  });
});