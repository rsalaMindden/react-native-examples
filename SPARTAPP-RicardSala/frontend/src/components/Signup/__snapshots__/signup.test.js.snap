// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Signup component When it is render Then should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#595260",
        "flex": 1,
      },
      Object {
        "alignItems": "center",
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="loginPageButton"
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/spartappYELLOW.png",
        }
      }
      style={
        Object {
          "height": 31,
          "marginTop": 54,
          "width": 146,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2C2E43",
        "borderColor": "#FFD523",
        "borderRadius": 100,
        "borderWidth": 5,
        "height": 180,
        "justifyContent": "center",
        "marginTop": 15,
        "width": 180,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/userDefaultPNG.png",
        }
      }
      style={
        Object {
          "bottom": 10,
          "height": 140,
          "width": 120,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "#FFD523",
      }
    }
  >
     editar
  </Text>
  <View
    style={
      Array [
        Object {
          "alignSelf": "flex-start",
          "marginLeft": 14,
          "marginTop": 10,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "elevation": 15,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 7,
            "width": 0,
          },
          "shadowOpacity": 0.43,
          "shadowRadius": 9.51,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFD523",
            },
          ]
        }
      >
        nombre
      </Text>
      <TextInput
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="User name"
        placeholderTextColor="#B2B1B9"
        style={
          Array [
            Object {
              "backgroundColor": "#2C2E43",
              "borderRadius": 18,
              "fontSize": 24,
              "height": 67,
              "marginBottom": 20,
              "width": 300,
            },
            Object {
              "color": "#B2B1B9",
            },
          ]
        }
        testID="usernameInput"
        textAlign="center"
        textContentType="name"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "elevation": 15,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 7,
            "width": 0,
          },
          "shadowOpacity": 0.43,
          "shadowRadius": 9.51,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFD523",
            },
          ]
        }
      >
        email
      </Text>
      <TextInput
        autoCapitalize="none"
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="example@example.com"
        placeholderTextColor="#B2B1B9"
        style={
          Array [
            Object {
              "backgroundColor": "#2C2E43",
              "borderRadius": 18,
              "fontSize": 24,
              "height": 67,
              "marginBottom": 20,
              "width": 300,
            },
            Object {
              "color": "#B2B1B9",
            },
          ]
        }
        testID="emailInput"
        textAlign="center"
        textContentType="emailAddress"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "elevation": 15,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 7,
            "width": 0,
          },
          "shadowOpacity": 0.43,
          "shadowRadius": 9.51,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFD523",
            },
          ]
        }
      >
        password
      </Text>
      <TextInput
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="password"
        placeholderTextColor="#B2B1B9"
        secureTextEntry={true}
        style={
          Array [
            Object {
              "backgroundColor": "#2C2E43",
              "borderRadius": 18,
              "fontSize": 24,
              "height": 67,
              "marginBottom": 20,
              "width": 300,
            },
            Object {
              "color": "#B2B1B9",
            },
          ]
        }
        testID="passwordInput"
        textAlign="center"
        textContentType="newPassword"
        value=""
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
        Object {
          "bottom": 50,
          "flexDirection": "row",
          "justifyContent": "space-around",
        },
        Object {
          "marginTop": 110,
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2C2E43",
          "borderRadius": 15,
          "elevation": 15,
          "height": 41,
          "justifyContent": "center",
          "opacity": 1,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 7,
            "width": 0,
          },
          "shadowOpacity": 0.43,
          "shadowRadius": 9.51,
          "width": 160,
        }
      }
      testID="registerButton"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFD523",
            },
            Object {
              "fontSize": 20,
              "fontWeight": "600",
              "justifyContent": "center",
              "marginLeft": 10,
              "marginRight": 10,
            },
          ]
        }
      >
        REGISTER
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
